dataplane/burst.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <rte_ethdev.h>
+#include <rte_mbuf.h>
+
+#include "common.h"
+#include "common/type.h"
+
+namespace dpdk
+{
+
+class BurstSpan
+{
+	rte_mbuf** first_;
+	uint16_t len_ = 0;
+
+public:
+	BurstSpan(rte_mbuf** first, uint16_t length) :
+	        first_{first}, len_{length}
+	{
+	}
+	bool empty() { return len_ == 0; }
+	rte_mbuf** begin()
+	{
+		return first_;
+	}
+	rte_mbuf** end()
+	{
+		return first_ + len_;
+	}
+	uint16_t Free()
+	{
+		if (empty())
+			return 0;
+		rte_pktmbuf_free_bulk(first_, len_);
+		auto freed = len_;
+		len_ = 0;
+		return freed;
+	}
+	uint64_t Bytes()
+	{
+		uint64_t bytes = 0;
+		for (auto& mbuf : *this)
+		{
+			bytes += rte_pktmbuf_pkt_len(mbuf);
+		}
+		return bytes;
+	}
+	uint16_t Packets() { return len_; }
+	BurstSpan& Tx(tPortId port_id, tQueueId queue_id)
+	{
+		if (empty())
+			return *this;
+		auto txSize = rte_eth_tx_burst(port_id, queue_id, first_, len_);
+		len_ -= txSize;
+		first_ += txSize;
+		return *this;
+	}
+	BurstSpan& Tx(tPortId port_id, tQueueId queue_id, uint64_t& opackets, uint64_t& obytes)
```

Comment:
This may be confusing as the function does not "return" `opackets` and `obytes` but increases them

Reply:
Rewrote `Burst`, ditched `BurstSpan`